src/CommunitiesList.cpp

```diff
@@ -193,3 +254,40 @@ CommunitiesList::roomList(const QString &id) const
 
         return {};
 }
+
+void
+CommunitiesList::sortEntries()
+{
+        std::vector<CommunitiesListItem *> header;
+        std::vector<CommunitiesListItem *> communities;
+        std::vector<CommunitiesListItem *> tags;
+        std::vector<CommunitiesListItem *> footer;
+        // remove all the contents and sort them in the 4 vectors
+        for (auto &entry : communities_) {
+                CommunitiesListItem *item = entry.second.data();
+                contentsLayout_->removeWidget(item);
+                // world is handled separately
+                if (entry.first == "world")
+                        continue;
+                // sort the rest
+                if (item->is_tag())
+                        if (entry.first == "tag:m.favourite")
+                                header.push_back(item);
+                        else if (entry.first == "tag:m.lowpriority")
+                                footer.push_back(item);
+                        else
+                                tags.push_back(item);
+                else
+                        communities.push_back(item);
+        }
+
+        contentsLayout_->insertWidget(contentsLayout_->count() - 1, communities_["world"].data());
+        for (auto item : header)
```

Comment:
Are you referring to the fact that `contentsLayout_->insertWidget(contentsLayout_->count() - 1, item);` is a long call and that it implies a lot of repetition?

I realized I could simply remove the final "Stretch" item and insert it back in the layout at the end, rather than insert all widgets "one before the end". Would that be okay too?

Reply:
Yes, I was referring to that repetition. I'm fine with any solution that doesn't have a lot of duplication.